### PR TITLE
Remove Linesearcher.Finished()

### PR DESCRIPTION
**Pull request by vladimir-ch**
* Removes `Linesearcher.Finished()`
* Changes the order of return values from `Linesearcher.Iterate()`. Operation is the main value returned, so put it first.
* Slightly cleans up `Backtracking` and `Bisection`
* And tries to update the docs for `Linesearcher` ...

@btracey Please take a look

### CODE REVIEW

File: backtracking.go [hunk 1]
```diff
@@ -57,17 +59,27 @@ func (b *Backtracking) Init(f, g float64, step float64) Operation {
...
 	b.stepSize *= b.Decrease
 	if b.stepSize < minimumBacktrackingStepSize {
-		return 0, NoOperation, ErrLinesearchFailure
+		return b.next(NoOperation, ErrLinesearchFailure)
 	}
-	return b.stepSize, FuncEvaluation, nil
+	return b.next(FuncEvaluation, nil)
+}
+
+func (b *Backtracking) next(op Operation, err error) (Operation, float64, error) {
```
Comment (btracey) [inline backtracking.go]: This function is simple and only happens twice. I think it's better to just put this in line.

File: interfaces.go [hunk 2]
```diff
@@ -65,24 +65,35 @@ type Statuser interface {
 // (approximate) minimum of the objective function along the search direction
 // dir_k starting at the most recent location x_k, i.e., it tries to minimize
 // the function
+//
```
Comment (btracey) [inline interfaces.go]: Don't need these newlines.

**Comment by btracey** [comment]
Nice. LGTM.

File: backtracking.go (hunk 1 above)
Comment (vladimir-ch) [inline backtracking.go]: Three times, but nevermind, inlined.

File: interfaces.go (hunk 2 above)
Comment (vladimir-ch) [inline interfaces.go]: I like the formulas to have some margin around them, similarly to mathematical formulas in papers. It looks better to me and the formula stands out, which is often good, for example in this case we refer to it at several places below. But ok, I removed the newlines.

**Pull request merged**